apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs

```diff
@@ -112,7 +118,10 @@ private static SpannerSettings CreateSpannerSettingsWithVersionHeader()
         internal Task<SessionPool> AcquireSessionPoolAsync(SpannerClientCreationOptions options)
         {
             GaxPreconditions.CheckNotNull(options, nameof(options));
-            var targetedPool = _targetedPools.GetOrAdd(options, key => new TargetedPool(this, key));
+            // If we're using the emulator, we need to apply special options. It's friendly to do this by default rather
+            // than requiring the client to set it up.
+            var poolOptions = options.UsesEmulator ? s_emulatorSessionPoolOptions : SessionPoolOptions;
```

Comment:
Good question - probably one for @skuruppu more than me. (I'm kinda hoping the need for custom options goes away...)

Reply:
It depends on if the users rely on these additional options when they're trying to use the emulator. For example, timeouts may not matter so much since the emulator will be able to return a result a lot sooner than the backend.

I didn't realize there were other options. The default pool uses an [empty options](https://github.com/googleapis/google-cloud-dotnet/blob/d736cdf2e725ca827d6bbb7b001947f28c016a2f/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs#L59) right?

Reply:
Emulator team said they don't plan to provide support for concurrent transactions anytime soon. Sorry :(

But this does raise an interesting question about the session pool in C#. We haven't needed a session pool configuration like this for any of the other languages when running tests against the emulator. What is the difference in C#? In most languages we have the 0.2 write fraction so the pool will initialize with 20% of the sessions having called BeginTransaction. This hasn't caused any issues with the emulator.

I need to take a look through https://github.com/googleapis/google-cloud-dotnet/blob/master/apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs to understand the pool initialization better.

Reply:
The emulator team confirmed that having multiple sessions calling BeginTransaction shouldn't be an issue with the emulator. A session has to have a read/write called on it which causes the database to get locked. Once the db is locked, the emulator will return the "Transaction Y aborted due to active transaction X. The emulator only supports one transaction at a time." error.

Reply:
The "empty options" really just means "default options" - see https://googleapis.dev/dotnet/Google.Cloud.Spanner.Data/latest/configuration.html#session-pool-options-explained for details.
I think we can probably manage without merging options for right now, but should revisit later.

As for why only C# needs this - let's investigate that internally.